ui/src/pages/users.vue

```diff
@@ -262,6 +342,61 @@ const autofill = ref({
   cas_id: "",
 });
 
+// New variables for delete confirmation modals
+const isDeleteModalVisible = ref(false);
+const isUsernamePromptVisible = ref(false);
+const usernameConfirmation = ref("");
+//const deleteOption = ref(null);
+const isHardDelete = ref(false);
+const impactDetails = ref(null);
```

Comment:
If you are setting `impactDetails` to an array, then it makes sense to also initialize it as an array, like this:

`const impactDetails = ref([])`

Also, if you initialize this as an array, all the places where you are checking for `impactDetails` to be `null` will need to be replaced with `impactDetails.length > 0` (within the `<template>` blocks) or `impactDetails.value.length > 0` (within the `<script />` blocks).

And the places where you are setting `impactDetails` to null will need to be changed, so that it is set to `[]` instead.